src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -1290,4 +1323,43 @@ protected void operationProcessed() {
                 threadPool, indexingService, indexSettingsService, warmer, store, deletionPolicy, translog, mergePolicyProvider, mergeScheduler,
                 mapperAnalyzer, similarityService.similarity(), codecService, failedEngineListener, translogRecoveryPerformer);
     }
+
+    private final AbstractRefCounted counter = new AbstractRefCounted("foo") {
+        @Override
+        protected void closeInternal() {
+            closeEngine();
```

Comment:
hmm why are we closing the engine later? I think we should just close it on IndexShard#close